src/ServiceControl.Monitoring/Http/Diagrams/TimingsAggregator.cs

```diff
@@ -39,6 +44,23 @@ public IEnumerable<MonitoredEndpoint> AggregateIntoLogicalEndpoints()
                     CriticalTime = AggregateTimings(instanceCriticalTime)
                 };
 
+                var queueLength = queueLengths.FirstOrDefault(ql => ql.Key == endpointName);
```

Comment:
Should this be a TryGet on the dictionary rather than scan for key, check value for null?